airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -113,17 +115,20 @@ def read(
                     continue
                 try:
                     timer.start_event(f"Syncing stream {configured_stream.stream.name}")
+                    self._emit_stream_status(configured_stream, AirbyteStreamStatus.STARTED, None)
                     yield from self._read_stream(
                         logger=logger,
                         stream_instance=stream_instance,
                         configured_stream=configured_stream,
                         state_manager=state_manager,
                         internal_config=internal_config,
                     )
+                    self._emit_stream_status(configured_stream, AirbyteStreamStatus.STOPPED, True)
                 except AirbyteTracedException as e:
                     raise e
                 except Exception as e:
                     logger.exception(f"Encountered an exception while reading stream {configured_stream.stream.name}")
+                    self._emit_stream_status(configured_stream, AirbyteStreamStatus.STOPPED, False)
```

Comment:
this message won't be yielded.

Option 1:
1. `_emit_stream_status` returns `stream_status_as_airbyte_message(stream, stream_status, successful)`
2. change these lines to yield the output of `_emit_stream_status` (`yield self._emit_stream_status(configured_stream, AirbyteStreamStatus.STOPPED, True)`)

Option 2:
change this line to `stream_status_as_airbyte_message(configured_stream, AirbyteStreamStatus.STOPPED, True)` and forget about the `_emit_stream_status` method

Reply:
@girarda Do you have a preference between the two options?  I'm inclined to implement option 1, unless you feel strongly about this.

Reply:
sounds good to me @jdpgrailsdev

Reply:
> sounds good to me @jdpgrailsdev

@girarda I didn't really understand the difference between options 1 and 2.  After looking at the code again, I went with option 2, as the method really didn't add any value anymore.